pkg/controllers/resources/nodes/syncer_test.go

```diff
@@ -167,10 +176,104 @@ func TestSyncBothExist(t *testing.T) {
 			virtualNodeExists: true,
 			expectNoVNode:     true,
 		},
+		{
+			name:                    "Label Matched and enforceNodeSelector false - expect node to be synced from NodeSelector",
+			virtualNodeExists:       true,
+			syncFromHostLabel:       map[string]string{"test": "true"},
+			hostLabel:               map[string]string{"test": "true"},
+			virtualFinalAnnotations: map[string]string{translate.ManagedLabelsAnnotation: "test"},
+			virtualFinalLabels:      map[string]string{"test": "true"},
+		},
+		{
+			name:              "Label Not Matched and enforceNodeSelector true - expect node not to be synced",
+			virtualNodeExists: true,
+			syncFromHostLabel: map[string]string{"test": "true"},
+			expectNoVNode:     true,
+		},
+		{
+			name:               "Clear Nodes",
+			virtualNodeExists:  true,
+			withVirtualPod:     true,
+			clearImage:         true,
+			imagesPhysicalNode: []corev1.ContainerImage{{Names: []string{"ghcr.io/jetpack/calico"}}},
+		},
+		{
+			name:               "Don't Clear Nodes",
+			virtualNodeExists:  true,
+			withVirtualPod:     true,
+			clearImage:         false,
```

Comment:
Since the default is false, is this needed?

Reply:
just wanted to be explicit, but yeah it's not really needed